### PR TITLE
route progress test added

### PR DESCRIPTION
**Pull request by cammace**

### CODE REVIEW

**Comment by cammace** [comment]
This PR adds a bunch of test and fixes up the RouteProgress classes (cleaned up code and fixed minor bugs). @zugaldia can you review?

**Comment by zugaldia** [comment]
@cammace I believe you were gonna introduce some changes to this PR (deltas vs rounding). Let me know when that lands and happy to 👀  the PR.

File: navigation/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/BaseTest.java [hunk 1]
```diff
@@ -10,8 +10,9 @@
 
 public class BaseTest {
   public static final double DELTA = 1E-10;
-  private static final String BASE_PATH = "/res/";
+  public static final double LARGE_DELTA = 0.1;
```
Comment (cammace) [inline navigation/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/BaseTest.java]: @zugaldia added a larger delta to offset the difference between API and calculated distances.

**Comment by cammace** [comment]
I've also added autoValues inside this PR to reinforce the immutable nature of RouteProgress. Ready for a review @zugaldia.

**Pull request merged**